functions.sql

```diff
@@ -56,8 +57,8 @@ SELECT
 					"access"), TRUE)
 		WHEN highway = 'path' THEN
 			CASE carto_path_type(bicycle, horse)
-				WHEN 'cycleway' THEN carto_int_access(COALESCE(NULLIF(bicycle, 'unknown'), "access"), FALSE)
-				WHEN 'bridleway' THEN carto_int_access(COALESCE(NULLIF(horse, 'unknown'), "access"), FALSE)
+				WHEN 'cycleway' THEN carto_int_access(bicycle, FALSE)
```

Comment:
I don't think this is a good idea.

The simplification you introduce here relies on a specific characteristic of the way `carto_path_type()` operates right now - but which does not inherently result from the tagging practice in OSM per se.

We could, for example, in the future decide to have `carto_path_type()` return `cycleway` on `highway=path` + `foot=no` + `horse=no` - and then your simplified logic would not be correct - i.e. you need `access` as a fallback.  Keeping the logic uniform across the different `highway=*` types also facilitates easier understanding by style developers.

Reply:
Happy to revert to an agreed position. I agree there is a symmetry and logic to the original more general formulation, but equally there's a logic to saying and documenting that promotion of path is contingent on bicycle / horse being set.